src/domain/flow-system/src/repos/flow/flow_event_store.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use event_sourcing::EventStore;
+use opendatafabric::DatasetID;
+
+use crate::*;
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+pub trait FlowEventStore: EventStore<FlowState> {
+    /// Generates new unique flow identifier
+    fn new_flow_id(&self) -> FlowID;
+
+    /// Returns the last dataset flow of certain type
+    fn get_last_specific_dataset_flow(
```

Comment:
I see you have indulged me in replacing the word "specific" in the FlowSystem trait with `of_type`. Should we do the same here for consistency?

Reply:
My bad, should have renamed both parts.